scripts/runtime/VibeRuntime.Common.ps1

```
@@ -164,27 +185,683 @@ function Get-VibeNestedPropertySafe {
+    return [pscustomobject]@{
+        mode = [string]$mode
+        source = 'host_decision'
+        reason = if ((Test-VibeObjectHasProperty -InputObject $rawDecision -PropertyName 'reason') -and -not [string]::IsNullOrWhiteSpace([string]$rawDecision.reason)) { [string]$rawDecision.reason } else { 'Host supplied an explicit structured code-task TDD decision.' }
+        exception = if ((Test-VibeObjectHasProperty -InputObject $rawDecision -PropertyName 'exception') -and -not [string]::IsNullOrWhiteSpace([string]$rawDecision.exception)) { [string]$rawDecision.exception } else { $null }
+    }
+}
+
+function Resolve-VibeCodeTaskTddDecision {
+    param(
+        [AllowNull()] [object]$HostDecision = $null,
+        [Parameter(Mandatory)] [string]$Task,
```

PSScriptAnalyzer (1.25.0)

[warning] The parameter 'Task' has been declared but not used.

(PSReviewUnusedParameter)